core/src/test/scala/pureconfig/ConfigConvertChecks.scala

```diff
@@ -23,7 +23,7 @@ trait ConfigConvertChecks { this: FlatSpec with Matchers with GeneratorDrivenPro
    * because, for instance, multiple representation of `t: T` are possible. Use [[checkRead()]] for those
    * representations.
    */
-  def checkArbitrary[T](implicit cc: ConfigConvert[T], arb: Arbitrary[T], tag: ClassTag[T]): Unit =
+  def checkArbitrary[T](implicit cc: ConfigConvert[T], arb: Arbitrary[T], tag: ClassTag[T], equality: Equality[T]): Unit =
     it should s"read an arbitrary ${tag.runtimeClass.getCanonicalName}" in forAll {
       (t: T) =>
         val result = cc.from(cc.to(t))
```

Comment:
@derekmorr Yeah, I wondered about the same thing when I was in this neighborhood recently. My vote would be to remove it. 

If we want to get separate line numbers reported for left vs right as compared to the equality failing, I'd do:
```
val value = result.right.value
value shouldEqual t
```